### PR TITLE
[3/N] Refactor FR script - Add a processor module

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133933
* #133929
* #133927

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o @xmfan

### CODE REVIEW

File: tools/flight_recorder/components/processor.py [hunk 1]
```diff
@@ -0,0 +1,899 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (fduwjj) [inline tools/flight_recorder/components/processor.py:1]: Will further refactor this class more.

**Review by c-p-i-o — APPROVED** [review]
Nice!

**Comment by fduwjj** [comment]
#suppress-bc-linter

**Pull request closed without merging**